lib/connextclient/ConnextClient.ts

```diff
@@ -0,0 +1,491 @@
+  private ethProviderUrl: string;
+  private nodeUrl: string;
+  private unitConverter: UnitConverter;
+
+  private client: IConnextClient | undefined;
+  private wallet: ConnextWallet | undefined;
+
+  /**
+   * Creates a connext client.
+   */
+  constructor({
+    config,
+    logger,
+    unitConverter,
+    currencyInstances,
+  }: {
+    unitConverter: UnitConverter;
+    config: ConnextClientConfig;
+    currencyInstances: CurrencyInstance[],
+    logger: Logger;
+  }) {
+    super(logger, config.disable);
+
+    this.ethProviderUrl = config.ethProviderUrl;
+    this.nodeUrl = config.nodeUrl;
+    this.unitConverter = unitConverter;
+    this.setTokenAddresses(currencyInstances);
+  }
+
+  public get minutesPerBlock() {
+    return 0.25; // 15 seconds per block target
+  }
+
+  public get label() {
+    return 'Connext';
+  }
+
+  /**
+   * Checks for connectivity and gets our Connext account address
+   */
+  public initSpecific = async () => {
+    if (!this.wallet) {
+      throw errors.CONNEXT_WALLET_NOT_INITIATED;
+    }
+    if (this.wallet) {
+      await this.initConnextClient();
+      this.setStatus(ClientStatus.Initialized);
+      this.emit('initialized');
+      await this.verifyConnectionWithTimeout();
+    }
+  }
+  /**
+   * Initiates wallet for the Connext client
+   */
+  public initWallet = async (seedMnemonic: string[]) => {
+    this.wallet = new ConnextWallet(seedMnemonic.join(' '));
+  }
+
+  public initConnextClient = async () => {
+    if (!this.wallet) {
+      throw errors.CONNEXT_WALLET_NOT_INITIATED;
+    }
+    const wallet = this.wallet;
+    const client = await connext.connect({
+      ethProviderUrl: this.ethProviderUrl,
+      nodeUrl: this.nodeUrl,
+      xpub: wallet.xpub,
+      keyGen: (index: string) => wallet.keyGen(index),
+      store: new ConnextStore(new FileStorage()),
+    });
+    this.client = client;
+    return client;
+  }
+  /**
+   * Associate connext with currencies that have a token address
+   */
+  private setTokenAddresses = (currencyInstances: CurrencyInstance[]) => {
+    currencyInstances.forEach((currency) => {
+      if (currency.tokenAddress) {
+        this.tokenAddresses.set(currency.id, currency.tokenAddress);
+      }
+    });
+  }
+
+  public totalOutboundAmount = (): number => {
+    // assume MAX_AMOUNT since Connext will re-collaterize accordingly
+    return MAX_AMOUNT;
+  }
+
+  public maxChannelOutboundAmount = (): number => {
+    // assume MAX_AMOUNT since Connext will re-collaterize accordingly
+    return MAX_AMOUNT;
+  }
+
+  public maxChannelInboundAmount = (): number => {
+    // assume MAX_AMOUNT since Connext will re-collaterize accordingly
+    return MAX_AMOUNT;
+  }
+
+  protected updateCapacity = async () => {
+    try {
+      const channelBalancePromises = [];
+      for (const [currency] of this.tokenAddresses) {
+        channelBalancePromises.push(this.channelBalance(currency));
+      }
+      await Promise.all(channelBalancePromises);
```

Comment:
It looks like this is currently pulling back all the channel balances, but it doesn't do anything with the data it receives. Would it make sense to set this as our max inbound/outbound amounts?